lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -623,6 +623,34 @@ struct SToR1 : public OpRewritePattern<quake::SOp> {
   }
 };
 
+// quake.s<adj> target
+// ─────────────────────────────────
+// (quake.z * quake.s) target
```

Comment:
Isn't this from Pradnya's PR? I changed this comment there. Might need to rebase?

Reply:
This decomposition pattern was removed since the existing `SToR1` was sufficient and this added additional gates in basis conversion. Again, probably will go away with rebase, making a note here to confirm that this change is not required.

Reply:
updated